burn-book/src/import/pytorch-model.md

```diff
@@ -0,0 +1,215 @@
+
+       // Save the model record to a file.
+       let recorder = NamedMpkFileRecorder::<FullPrecisionSettings>::default();
+       recorder
+           .record(record, "MY_FILE_OUTPUT_PATH".into())
+           .expect("Failed to save model record");
+   }
+
+   /// Load the model from the file in your source code (not in build.rs or script).
+   fn load_model() -> Net::<Backend> {
+       let device = Default::default();
+       let record = NamedMpkFileRecorder::<FullPrecisionSettings>::default()
+           .load("./MY_FILE_OUTPUT_PATH".into(), &device)
+           .expect("Should decode state successfully");
+
+       Net::<Backend>::new_with(record)
+   }
+   ```
+
+## Fixing and troubleshooting
+
+### Adjusting the source model architecture
+
+If your target model differs structurally from the model you exported, `PyTorchFileRecorder` allows
+changing the attribute names and the order of the attributes. For example, if you exported a model
+with the following structure:
+
+```python
+class ConvModule(nn.Module):
+    def __init__(self):
+        super(ConvModule, self).__init__()
+        self.conv1 = nn.Conv2d(2, 2, (2,2))
+        self.conv2 = nn.Conv2d(2, 2, (2,2), bias=False)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.conv2(x)
+        return x
+
+class MyModel(nn.Module):
+    def __init__(self):
+        super(MyModel, self).__init__()
+        self.conv = ConvModule()
+
+    def forward(self, x):
+        x = self.conv(x)
+        return x
+```
+
+But you need to import it into a model with the following structure:
+
+```rust
+#[derive(Module, Debug)]
+pub struct Net<B: Backend> {
+    conv1: Conv2d<B>,
+    conv2: Conv2d<B>,
+}
+```
+
+Which produces the following weights structure (viewed in
+[Netron](https://github.com/lutzroeder/netron)):
+
+![image alt >](./key_remap.svg)
+
+You can use the `PyTorchFileRecorder` to change the attribute names and the order of the attributes
+by specifying a regular expression (See
+[regex::Regex::replace](https://docs.rs/regex/latest/regex/struct.Regex.html#method.replace)) to
+match the attribute name and a replacement string in `LoadArgs`:
+
+```rust
+let device = Default::default();
+let load_args = LoadArgs::new("tests/key_remap/key_remap.pt".into())
+    // Remove "conv" prefix, e.g. "conv.conv1" -> "conv1"
+    .with_key_remap("conv\\.(.*)", "$1");
+
+let record = PyTorchFileRecorder::<FullPrecisionSettings>::default()
+    .load(load_args, &device)
+    .expect("Should decode state successfully");
+
+let model = Net::<Backend>::new_with(record);
+```
+
+### Loading the model weights to a partial model
+
+`PyTorchFileRecorder` supports loading parts of the model weights to a partial model. For example,
+if your source model contains an encoder and a decoder, but you only want to load the encoder
+weights, you can define the encoder in Burn and load the encoder weights to the encoder,
+ignoring the decoder weights.
```

Comment:
In my opinion, structuring this part thourgh bullet points could make the concepts catchable at first glance

Reply:
I tried using list but didn't work for me. I reworded the previous one. If you have alternative in mind, please suggest it.

Reply:
Let me try, something like:

_For example, if your source model is composed of an encoder and a decoder, you can decide to load only the encoder weights, but not the decoder weights. To achieve that, you just have to define the encoder model in Burn and load its weights._